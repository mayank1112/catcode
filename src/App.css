body {
  margin: 12px;
}

button {
  cursor: pointer;
  background: transparent;
  border: 0;
  position: relative;
  width: 120px;
  height: 120px;
  text-align: center;
  margin: 4px;
}

