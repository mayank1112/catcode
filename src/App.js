import React, { Fragment, useEffect, useState } from 'react';
import axios from 'axios';
import './App.css';
import { bagUrl, clouderUrl } from './utils/paths';
import { ERROR_MESSAGE, SUCCESS_MESSAGE } from './constants/lang';
import Header from './components/Header/Header.jsx';
import { CATS_GROUP_SIZE } from './constants/apiConstants';
import { updateSelectedCatsValue } from './utils/generalUtils';
import CatsApp from './components/CatsApp/CatsApp';

function App() {
  const [cats, setCats] = useState([]);
  const [selectedCats, setSelectedCats] = useState([]);
  const [groupedCats, setGroupedCats] = useState([]);
  const [successMessage, setSuccessMessage] = useState();
  const [errorMessage, setErrorMessage] = useState();
  const [selectedCatsCount, setSelectedCatsCount] = useState(0);

  useEffect(() => {
    const fetchCats = async () => {
      const response = await axios(bagUrl());
      setCats(response.data.cats);
    }
    fetchCats();
  }, []);

  const checkCatsCompatibility = async catIds => {
    try {
      await axios(clouderUrl(catIds));
      setSuccessMessage(SUCCESS_MESSAGE);
      setGroupedCats(catIds);
    } catch (error) {
      setErrorMessage(ERROR_MESSAGE);
    } finally {
      setSelectedCats([]);
      setSelectedCatsCount(0);
    }
  };

  const selectCat = async catId => {
    if (!groupedCats.includes(catId)) {
      const selectedCatsValue = updateSelectedCatsValue(selectedCats, catId);
      if (selectedCatsValue.length === CATS_GROUP_SIZE) {
        await checkCatsCompatibility(selectedCatsValue);
      } else {
        setSelectedCatsCount(selectedCatsValue.length);
        setSelectedCats(selectedCatsValue);
      }
    }
  };
  return (
    <Fragment>
      <Header successMessage={successMessage} errorMessage={errorMessage} />
      <CatsApp
        cats={cats}
        groupedCats={groupedCats}
        selectedCats={selectedCats}
        selectCat={selectCat}
        selectedCatsCount={selectedCatsCount}
      />
    </Fragment>
  );
}

export default App;
