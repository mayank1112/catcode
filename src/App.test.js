import Enzyme, { shallow } from 'enzyme';
import App from './App.js';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

it('<App> component should match snapshot', () => {
    const component = shallow(<App />);
    expect(toJson(component)).toMatchSnapshot();
});
