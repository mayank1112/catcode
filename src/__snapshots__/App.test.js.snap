// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> component should match snapshot 1`] = `
<Fragment>
  <Header />
  <CatsApp
    cats={Array []}
    groupedCats={Array []}
    selectCat={[Function]}
    selectedCats={Array []}
    selectedCatsCount={0}
  />
</Fragment>
`;
